securesignals-ima-dev-app/src/main/java/com/uid2/dev/MainActivity.java

```diff
@@ -0,0 +1,278 @@
+                                        // AdEventType.CONTENT_PAUSE_REQUESTED is fired when you
+                                        // should pause your content and start playing an ad.
+                                        pauseContentForAds();
+                                        break;
+                                    case CONTENT_RESUME_REQUESTED:
+                                        // AdEventType.CONTENT_RESUME_REQUESTED is fired when the ad
+                                        // you should play your content.
+                                        resumeContent();
+                                        break;
+                                    case ALL_ADS_COMPLETED:
+                                        // Calling adsManager.destroy() triggers the function
+                                        // VideoAdPlayer.release().
+                                        adsManager.destroy();
+                                        adsManager = null;
+                                        break;
+                                    case CLICKED:
+                                        // When the user clicks on the Learn More button, the IMA SDK fires
+                                        // this event, pauses the ad, and opens the ad's click-through URL.
+                                        // When the user returns to the app, the IMA SDK calls the
+                                        // VideoAdPlayer.playAd() function automatically.
+                                        break;
+                                    default:
+                                        break;
+                                }
+                            }
+                        });
+                    AdsRenderingSettings adsRenderingSettings =
+                        ImaSdkFactory.getInstance().createAdsRenderingSettings();
+                    adsManager.init(adsRenderingSettings);
+                }
+            });
+
+        // When the play button is clicked, request ads and hide the button.
+        playButton = findViewById(R.id.playButton);
+        playButton.setOnClickListener(
+            view -> {
+                videoPlayer.setVideoPath(SAMPLE_VIDEO_URL);
+                requestAds(SAMPLE_VAST_TAG_URL);
+                view.setVisibility(View.GONE);
+            });
+    }
+
+    private void loadUID2Identity() {
+        InputStream is = getResources().openRawResource(R.raw.uid2identity);
+        StringBuilder text = new StringBuilder();
+
+        try (BufferedReader br = new BufferedReader(new InputStreamReader(is))) {
+            String line;
+
+            while ((line = br.readLine()) != null) {
+                text.append(line);
+                text.append('\n');
+            }
+
+            String jsonString = text.toString();
+            JSONObject jsonObject = (JSONObject) new JSONTokener(jsonString).nextValue();
```

Comment:
This could be simplified by just using `new JSONObject(jsonString)`. You don't need the `JSONTokener`

Reply:
Nice!  Changed made.